docs/packages/TypeScript_ESLint.mdx

```diff
@@ -83,6 +107,56 @@ We **_strongly_** recommend declaring our plugin with the namespace `@typescript
 You may choose a different namespace - but note that currently [flat configs allow the same plugin to be registered, configured, and have duplicate reports under multiple namespaces](https://github.com/eslint/eslint/discussions/17766).
 :::
 
+## `config(...)`
+
+The `config` function is a [variadic](https://en.wikipedia.org/wiki/Variadic_function) [identity function](https://en.wikipedia.org/wiki/Identity_function) which is a fancy way of saying that it's a function that takes `1..n` inputs and returns the inputs unchanged. It can be used to quickly and easily provide types for your flat config file. By using this function you will get autocomplete and documentation for all config properties in addition to TypeScript errors should you provide invalid values.
```

Comment:
I was trying to make it explicit what the function was so that nobody would be confused as to what the function was.

I did end up linking out because I remember the first time I read "variadic" I had to google it and then I understood what it meant -- "oh it's a function with a spread parameter".

Ultimately my goal was to make it clear that `[...]` is the same as `config(...)`, and not `config([...])`. In some of the user issues people were unclear on the usage and referenced it as `config([...])` which isn't right.

"Variadic" is nice because it immediately defines it as `config(...)` which informs the rest of the description.

I think we can do better here though.